linux/deploy_vm/create_unattend_install_iso.yml

```diff
@@ -5,69 +5,175 @@
+
 - debug:
-    msg: "Will generate unattend install ISO file in: {{ generated_unattend_iso }}"
+    msg: "The unattend install ISO to be generated is {{ generated_unattend_iso }}"
+
+- name: "Create temp folder for generating unattend install ISO file"
+  file:
+    path: "{{ unattend_iso_cache }}"
+    mode: "0777"
+    state: directory
+
+- name: "Get user password hash"
+  set_fact:
+    vm_password_hash: "{{ vm_password | password_hash('sha512') }}"
 
-- name: "Set fact of the absolute path of unattend install config file"
+# For nvme boot disk, boot device name shoule be nvme0n1 instead of sda
+- name: "Set boot disk name"
   set_fact:
-    unattend_install_conf: "{{ main_playbook_path }}/autoinstall/{{ unattend_install_conf }}"
-- name: "Display the path of unattend install config file"
-  debug: var=unattend_install_conf
+    boot_disk_name: "{{ 'nvme0n1' if boot_disk_controller == 'nvme' else 'sda' }}"
 
+# Get SSH public key
+- include_tasks: ../utils/get_local_ssh_public_key.yml
+- name: "Get SSH public key"
+  set_fact:
+    ssh_public_key: "{{ lookup('file', ssh_public_key_file) }}"
+
+- name: "Create unattend config file from template"
+  template:
+    src: "{{ unattend_install_template }}"
+    dest: "{{ new_unattend_install_conf }}"
+    mode: "0644"
+
+# Add new user config in unattend install config file
 - block:
-  # For nvme boot disk, boot device name shoule be nvme0n1 instead of sda
-    - name: "Get the configured unattend install control file name"
+    # Set post script content
+    - name: "Set post script for adding new user to sudoers"
       set_fact:
-        unattend_file: "{{ unattend_install_conf.split('/')[-1] }}"
-    - name: "Copy original unattend file to /tmp"
-      copy:
-        src: "{{ unattend_install_conf }}"
-        dest: "/tmp/{{ unattend_file }}"
-        mode: 0644
-
-    - name: "Set new unattend install control file"
-      set_fact:
-        unattend_install_conf: "/tmp/{{ unattend_file }}"
-    - name: "Replace =sda with nvme device name"
-      replace:
-        path: "{{ unattend_install_conf }}"
-        regexp: "=sda"
-        replace: "=nvme0n1"
-    - name: "Replace /dev/sda with nvme device name"
-      replace:
-        path: "{{ unattend_install_conf }}"
-        regexp: "/dev/sda"
-        replace: "/dev/nvme0n1"
-  when: boot_disk_controller == 'nvme'
-
-- name: Create a guest OS unattend install iso file
+        post_script: "echo '{{ new_user }} ALL=(ALL) NOPASSWD:ALL' >/etc/sudoers.d/{{ new_user }}"
+
+    # Add new user for RHEL and its varieties
+    - block:
+        - name: "Add a new user {{ new_user }} in {{ unattend_os_type }} unattend install config file"
+          lineinfile:
+            path: "{{ new_unattend_install_conf }}"
+            insertafter: "^rootpw"
+            line: 'user --name={{ new_user }} --password={{ vm_password_hash }} --groups=root,wheel --iscrypted --gecos="{{ new_user }}"'
```

Comment:
We can use this in template file to avoid reconfigure the file:
```
{% if user_name != 'root' %}
user --name={{ user_name }} --password={{ vm_password_hash }} --groups=root,wheel --iscrypted --gecos="{{ new_user }}""
{% endif %}
```

Reply:
Good idea.